File: core/src/test/java/com/sys1yagi/swipe/core/tool/ColorConverterTest.java
package com.sys1yagi.swipe.core.tool;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class ColorConverterTest {

    @Test
    public void testToColorInt() throws Exception {
        {
            int color = ColorConverter.toColorInt("#000000");
            assertThat(color).isEqualTo(-16777216);
        }
        {
            int color = ColorConverter.toColorInt("#FFFFFF");
            assertThat(color).isEqualTo(-1);
        }
        {
            int color = ColorConverter.toColorInt("#ffffff");
            assertThat(color).isEqualTo(-1);
        }
        {
            int color = ColorConverter.toColorInt("ffffff");
            assertThat(color).isEqualTo(-1);
        }
        {
            int color = ColorConverter.toColorInt("#fff");
            assertThat(color).isEqualTo(-1);
        }
        {
            int color = ColorConverter.toColorInt("fff");
            assertThat(color).isEqualTo(-1);
        }
        {
            int color = ColorConverter.toColorInt("abc");
            assertThat(color).isEqualTo(-5588020);
        }
    }
}

File: core/src/test/java/com/sys1yagi/swipe/core/tool/ColorConverterTest.kt
package com.sys1yagi.swipe.core.tool

import org.junit.Test

import org.assertj.core.api.Assertions.assertThat

class ColorConverterTest {

    @Test
    @Throws(Exception::class)
    fun testToColorInt() {
        run {
            val color = ColorConverter.toColorInt("#000000")
            assertThat(color).isEqualTo(-16777216)
        }
        run {
            val color = ColorConverter.toColorInt("#FFFFFF")
            assertThat(color).isEqualTo(-1)
        }
        run {
            val color = ColorConverter.toColorInt("#ffffff")
            assertThat(color).isEqualTo(-1)
        }
        run {
            val color = ColorConverter.toColorInt("ffffff")
            assertThat(color).isEqualTo(-1)
        }
        run {
            val color = ColorConverter.toColorInt("#fff")
            assertThat(color).isEqualTo(-1)
        }
        run {
            val color = ColorConverter.toColorInt("fff")
            assertThat(color).isEqualTo(-1)
        }
        run {
            val color = ColorConverter.toColorInt("abc")
            assertThat(color).isEqualTo(-5588020)
        }
    }
}
